plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestStandardColumnMappings.java

```diff
@@ -0,0 +1,126 @@
+                {12, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123},
+        };
+    }
+
+    @Test(dataProvider = "invalidShortTimestamps")
+    public void testVerifyInvalidShortTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro, String expectedException)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
+        int finalPicosOfMicro = picosOfMicro + (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        assertThatThrownBy(() -> verifyShortTimestamp(epochMicros, finalPicosOfMicro, precision)).hasMessageContaining(expectedException);
+    }
+
+    @DataProvider
+    public Object[][] invalidShortTimestamps()
+    {
+        return new Object[][] {
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 100_000_000), 0, "Invalid value"},
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {1, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {2, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {3, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {4, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {5, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {6, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {7, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Precision must be less"},
+                {11, LocalDateTime.of(2018, 3, 25, 3, 17, 17), 1, "Precision must be less"},
+        };
+    }
+
+    @Test(dataProvider = "invalidLongTimestamps")
+    public void testVerifyInvalidLongTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro, String expectedException)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        int finalPicosOfMicro = picosOfMicro + (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        assertThatThrownBy(() -> verifyLongTimestamp(finalPicosOfMicro, precision)).hasMessageContaining(expectedException);
+    }
+
+    @DataProvider
+    public Object[][] invalidLongTimestamps()
+    {
+        return new Object[][] {
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 100_000_000), 0, "Precision must be greater"},
+                {6, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Precision must be greater"},
+                {7, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
+                {8, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123, "Invalid value"},
```

Comment:
If we want to be serious about testing the method, please add also tests before epoch.

Reply:
Done!